app/models/category_list.rb

```diff
@@ -144,7 +144,7 @@ def find_categories
 
   def prune_empty
     return if SiteSetting.allow_uncategorized_topics
-    @categories.delete_if { |c| c.uncategorized? && c.displayable_topics.blank? }
+    @categories.delete_if { |c| c.uncategorized? }
```

Comment:
Shouldn't this be an `or` instead of removing the condition?

```suggestion
    @categories.delete_if { |c| c.uncategorized?  || c.displayable_topics.blank? }
```

Reply:
I do not think so. Your solution would not display any empty categories, which I do not think is intended. Empty categories are rare, but possible in new forums.